src/mapping.rs

```diff
@@ -452,19 +452,47 @@ fn parse_proguard_record(bytes: &[u8]) -> (Result<ProguardRecord, ParseError>, &
 fn parse_proguard_header(bytes: &[u8]) -> Result<(ProguardRecord, &[u8]), ParseError> {
     let bytes = parse_prefix(bytes, b"#")?;
 
-    let (key, bytes) = parse_until(bytes, |c| *c == b':' || is_newline(c))?;
+    if bytes.starts_with(b" {") {
+        // Parse as JSON-like string
+        let (json_str, bytes) = parse_until(bytes, |c| *c == b'}')?;
+        let bytes = parse_prefix(bytes, b"}")?;
 
-    let (value, bytes) = match parse_prefix(bytes, b":") {
-        Ok(bytes) => parse_until(bytes, is_newline).map(|(v, bytes)| (Some(v), bytes)),
-        Err(_) => Ok((None, bytes)),
-    }?;
+        // Remove the curly braces and split the string to extract key and value
+        let json_str = json_str.trim_matches(|c| c == '{' || c == '}');
+        let parts: Vec<&str> = json_str.split(|c| c == ':' || c == '\"').collect();
 
-    let record = ProguardRecord::Header {
-        key: key.trim(),
-        value: value.map(|v| v.trim()),
-    };
+        if parts.len() < 10 {
+            return Err(ParseError {
+                line: bytes,
+                kind: ParseErrorKind::ParseError("Invalid JSON format"),
+            });
+        }
 
-    Ok((record, consume_leading_newlines(bytes)))
+        let key = parts[4].trim();
+        let value = parts[9].trim();
```

Comment:
That's quite dirty - I was considering using JSON Crate like `serde_json` but I had some borrowing issue I didn't quite get.

Reply:
I would be okay with going with `serde_json` with a derive for this. I assume the borrowing problem comes from the fact that Serde will deserialize things as either `String` or `Cow<'a, str>`, and that will mess with lifetimes?

Either way, even though the serialization is technically "arbitrary" JSON, I believe we can get away with just matching things as strictly as possible, especially since `sourceFile` is the only json-like header we are interested in.

Which essentially means that you should be able to use `parse_prefix(br#"{"id":"sourceFile","fileName":""#)` to skip over the whole `sourceFile` prefix, and then take the filename, skipping over the trailing `"}`.

Reply:
> Either way, even though the serialization is technically "arbitrary" JSON, I believe we can get away with just matching things as strictly as possible, especially since sourceFile is the only json-like header we are interested in.

Good idea, thanks! Addressed in f729b5551e7bc7917e34ade3056b5b6b60af3b8f